caches/infinispan/src/test/resources/bmunit/common/WriteCopyAndReadNewFile.btm

```diff
@@ -0,0 +1,21 @@
+RULE service copyFile
+CLASS FastLocalCacheProvider
+METHOD copy
+AT ENTRY
+IF TRUE
+DO
+    debug("<<<copyFile is waiting for service enter writeFile");
+    waitFor("service writeFile");
+    debug("<<<proceed with copyFile")
+ENDRULE
+
+RULE service writeFile
+CLASS FastLocalCacheProvider
+METHOD unlockWrite
+AT EXIT
```

Comment:
Using Partyline (or in this case, something that uses Partyline), it should be possible to read from a file *while* it's being written. So, you might be able to use AT ENTER (if that's the term) instead of EXIT. I'm not sure, it might require watching a method deeper in the call stack (JoinableFile CTOR for example, or even EXIT of JoinableFileManager.openOutputStream()), but we should find a way to test that we can read from a file at the same time we're writing to it.